docs/APP_DEVELOPMENT.md

```diff
@@ -29,69 +28,158 @@ An app typically has the structure:
-To change the state, modify ``window.ove.state.current``.
+### Handling state
 
-The current application state can be sent as a WebSocket broadcast by calling ``OVE.Utils.broadcastState()`` (with no arguments).
-This will update the current state on all clients that receive the message; you can register a callback function to be called when this change occurs using ``OVE.Utils.setOnStateUpdate(callbackFunctionName)``.
+The `window.ove.state` object provides a `window.ove.state.current` data structure to hold the current application state. You can decide what this should contain, given the particular needs of your application.
 
+The current application state (contents of `window.ove.state.current`) can be sent as a WebSocket broadcast by calling `OVE.Utils.broadcastState()` (with no arguments). This will update the current state on all clients that receive the message; you can register a callback function to be called when this change occurs using `OVE.Utils.setOnStateUpdate(callbackFunctionName)`.
 
-## Helper methods
+The `window.ove.state` object also provides two other methods `cache` and `load`, which can be used to cache the application state on the server and load it sometime later. These methods are internally called by the utility methods provided by `OVE.Utils` and therefore their use is limited to a few advanced use-cases.
+
+### Interpreting layout
+
+The `window.ove.layout` provides information useful to interpret the layout of the clients:
+
+1. window.ove.layout.x - Displacement along the x-axis relative to the top-left of the section in pixels
+2. window.ove.layout.y - Displacement along the y-axis relative to the top-left of the section in pixels
+3. window.ove.layout.w - Width of the client
+4. window.ove.layout.h - Height of the client
+5. window.ove.layout.section.w - Width of the section (or the total width of all clients)
+6. window.ove.layout.section.h - Height of the section (or the total height of all clients)
+7. window.ove.layout.offset.x - Displacement of top-left of the client along the x-axis relative to the top-left of the browser window in pixels
+8. window.ove.layout.offset.y - Displacement of top-left of the client along the y-axis relative to the top-left of the browser window in pixels
 
-``OVE.Utils`` provides a number of useful methods, such as ``OVE.Utils.getQueryParam()``
+While all of this information is available on a fully initialized viewer, the controller only has:
+
+1. window.ove.layout.section.w - Width of the section (or the total width of all clients)
+2. window.ove.layout.section.h - Height of the section (or the total height of all clients)
+
+### Communicating via WebSockets
+
+The `window.ove.socket` provides two functions `send` and `on` to send and receive messages using WebSockets:
+
+``` JavaScript
+window.ove.socket.on(function (message) {
+    // logic to interpret the message
+});
+window.ove.socket.send(message);
```

Comment:
Clarify whether `message` can be an object (in which case, it must presumably be JSON serializable), or must be a string.